### PR TITLE
fix user api

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2439

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java [hunk 1]
```diff
@@ -649,7 +649,33 @@ public ResponseEntity<String> sendCode(@RequestBody EmailSenderInfo data) {
...
 		}
-		return userdataService.sendCode(data.action, data.lang, dev);
+		PremiumUsersRepository.PremiumUser pu = usersRepository.findById(dev.userid);
+		if (pu == null) {
+			return ResponseEntity.badRequest().body("User not found");
+		}
+		return userdataService.sendCode(data.action, data.lang, pu);
+	}
+	
+	@PostMapping(path = {"/auth/send-code-to-new-email"})
```
Comment (RZR-UA) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java:659]: Should be protected against flooding PremiumUsersRepository with tons of emails.

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java [hunk 2]
```diff
@@ -648,7 +649,33 @@ public ResponseEntity<String> sendCode(@RequestBody EmailSenderInfo data) {
...
+	@ResponseBody
+	public ResponseEntity<String> sendCodeToNewEmail(@RequestBody EmailSenderInfo data, @RequestParam String email) {
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java:661]: Here we don't need ```EmailSenderInfo data``` later again will be same issue that fields to class could be added and affect security

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java [hunk 3]
```diff
@@ -663,16 +690,19 @@ public ResponseEntity<String> confirmCode(@RequestBody String code) {
 	
 	@PostMapping(path = {"/auth/change-email"})
 	@ResponseBody
-	public ResponseEntity<String> changeEmail(@RequestBody UserPasswordPost us, HttpServletRequest request) throws ServletException {
-		if (us.username != null) {
-			us.username = us.username.toLowerCase().trim();
-		}
-		if (emailSender.isEmail(us.username)) {
+	public ResponseEntity<String> changeEmail(@RequestBody Map<String, String> credentials, HttpServletRequest request) throws ServletException {
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java:693]: Could you use more transparent way? @RequestBody String username, @RequestBody String token?

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java [hunk 4]
```diff
@@ -211,12 +204,17 @@ public ResponseEntity<String> deleteAccount(@RequestParam String token, HttpServ
 
 	@PostMapping(path = { "/auth/activate" }, consumes = "application/json", produces = "application/json")
 	@ResponseBody
-	public ResponseEntity<String> activateMapUser(@RequestBody UserPasswordPost us, HttpServletRequest request)
-			throws ServletException, IOException {
-		ResponseEntity<String> res = userdataService.webUserActivate(us.username, us.token, us.password);
+	public ResponseEntity<String> activateMapUser(@RequestBody Map<String, String> credentials, HttpServletRequest request) throws ServletException {
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java:207]: Map is not better than class.

**Pull request merged**